common-pg/src/main/scala/com/socrata/pg/store/RollupManager.scala

```diff
@@ -140,6 +140,22 @@ class RollupManager(pgu: PGSecondaryUniverse[SoQLType, SoQLValue], copyInfo: Cop
     }
   }
 
+  def bumpVersionRollup(rollupInfo: RollupInfo, oldCopyInfo: Option[CopyInfo], newDataVersion: Long): Boolean = {
+    logger.info(s"Updating copy $copyInfo, rollup $rollupInfo")
+    time("update-rollup", "dataset_id" -> copyInfo.datasetInfo.systemId, "rollupName" -> rollupInfo.name) {
+      val newTableName = rollupTableName(rollupInfo, newDataVersion)
+      oldCopyInfo.filter(_.copyNumber == rollupInfo.copyInfo.copyNumber).foreach { ci =>
+        val oldTableName = rollupTableName(rollupInfo, ci.dataVersion)
+        using(pgu.conn.createStatement()) { stmt =>
+          val sql = s"ALTER TABLE ${oldTableName} RENAME TO ${newTableName}"
```

Comment:
Is this going to cause locking issues for any queries trying to use this table?  My understanding was renaming a table required an exclusive lock.   eg. if you add a new rollup that takes a while to build, and have already renamed another rollup in that transaction, won't that block read queries that try to access it?

Reply:
I combine all alter table statements into one and give it 60s timeout.  Other long but safe operations are all executed before alter tables.  If alter tables does time out due to lock, it will roll back and fall back to the slow method.
The slow method drops rollup tables immediately.  I wonder why the slow method would not cause similar exclusive lock issue (I haven't verified about that yet.)

Reply:
I'm surprised the existing method drops rollup tables immediately.

I still have concerns about what happens to queries reading from the rollups while this happens.

Reply:
Do you have a concrete example that will cause problem with the logic?

Reply:
1. have a query reading from the rollup, which takes a shared lock
2. have an alter blocking on a lock (or anything else after that but before commit)
3. when a second query tries to read from the rollup, it will now block trying to get a shared lock

Reply:
My puzzle about the existing immediate drop is solved.  It is trying to drop new version of rollup before creating the new version.  It isn't dropping the old version.

About your example, yes - it will block other read queries but only up to 60s and then it will fall back to the slow method.  I am okay to lower that number or better yet, make that configurable.

Reply:
Perhaps an alternative to renaming is `create table the_new_name like the_old_name; insert into the_new_name select * from the_old_name` would work?  It would copy all the data, but it wouldn't need to _recompute_ all the data...

Reply:
@rjmac that would definitely help for the initially designed use case where rollups are small, but I worry that some of our current cases have rather large "rollups" that aren't really doing much aggregation of data so would still be almost as slow.  We may want to dig into if that worry is justified privately.

Oh, it would also need to create indexes too, as rollups currently get the full index treatment in an attempt to ensure you don't end up with queries being slower hitting an unindexed rollup.

Reply:
Another option is changing the naming scheme so there is a level of indirection so we don't have to have the rollup tables named for the data version, so they don't have to be renamed/recreated unless actual data changes, but I haven't thought that through.

Reply:
Even for < 10s?  This is especially useful for development I wish we have a mechanism as flexible as soql hints.

Reply:
I thought about the create table like and insert into.  That will be a half way compromise.

Reply:
how about a 10s timeout and falling back to create table like?

Reply:
I am not a big fan of adding multiple hard to tests codepaths.  It has repeatedly bit us in related areas in the past.

Reply:
I think it is well worth the speed gain.  But I am not insisting and remove the speediest path for now.